File: java/13514022_ITI/src/makhluk/Point.java
package makhluk;
/**
* @author Kristianto K./13514075
* Kelas Point. Untuk merepresentasi koordinat suatu objek
*/
public class Point {
    /** Posisi absis. */
    private int x;
    /** Posisi ordinat. */
    private int y;

    /** Ctor point.
    */
    public Point() {
       this.x = 0; this.y = 0;
    }
    /** Ctor point yang berparameter. x = _x dan y = _y
    *
    * @param a absis
    * @param b ordinat
    *
    */
    public Point(final int a, final int b) {
        this.x = a; this.y = b;
    }
    /** Setter point. x = _x dan y = _y
    *
    * @param a absis
    * @param b ordinat
    *
    */
    public final void setPoint(final int a, final int b) {
        this.x = a;
        this.y = b;
    }
    /** Getter absis.
    * @return int absis
    */
    public final int getX() {
        return this.x;
    }
    /** Getter ordinat.
    * @return int ordinat
    */
    public final int getY() {
        return this.y;
    }
    /** Menggerakkan point sebesar _x dan _y. x = x + _x dan y = y +_y.
    * @param a delta X
    * @param b delta Y
    */
    public final void movePoint(final int a, final int b) {
        this.x = a + this.x;
        this.y = b + this.y;
    }
}

File: cpp/2A - NONTHREAD - STL/Point.cpp
#include "Point.h"
using namespace std;

Point::Point(){
    x = 0;
    y = 0;
}
Point::Point(int _x,int _y){
    x = _x;
    y = _y;
}

void Point::SetPoint(int _x,int _y){
    x = _x;
    y = _y;
}
int Point::GetX(){ return x;}
int Point::GetY(){ return y;}
void Point::movePoint(int _x,int _y){
    x = _x + x;
    y = _y + y;
}
